README.md

```diff
@@ -1405,6 +1407,22 @@ Once enabled, you must configure Hystrix for each command and start the Hystrix
 
 See [Hystrix godoc](https://godoc.org/github.com/afex/hystrix-go/hystrix) for more info and [examples/hystrix](https://github.com/rs/rest-layer/blob/master/examples/hystrix/main.go) for a complete usage example with REST layer.
 
+## JSONSchema
+
+An incomplete but stil useful JSONSchema implementation that covers many common use cases for Schema. Goal is to try and match the draft-04 spec. Patches welcome.
+
+```go
+import "github.com/rs/rest-layer/schema/jsonschema"
+
+b := new(bytes.Buffer)
+enc := jsonschema.NewEncoder(b)
+if err := enc.Encode(myschema); err != nil {
+  return err
+}
+fmt.Println(b.String()) // Valid JSON Document describing the schema
+```
+
+
```

Comment:
@rs / @yanfali: Would it make sense to add a checklist for which types are supported atm? It might make it easier to contribute / know the state of the feature.

Here is a draft list based on what I can see:
```
[x] schema.Schema
[ ] schema.Bool
[ ] schema.Null
[x] schema.Float
[x] schema.Integer
[x] schema.String
[x] schema.Time
[ ] schema.URL
[ ] schema.IP
[ ] schema.Password
[x] schema.Array
[x] schema.Object
[ ] schema.Dict
[ ] schema.AllOf
[ ] schema.AnyOf
[ ] Custom validators
```